lib/handlers/hard-quote/handler.ts

```diff
@@ -57,6 +104,61 @@ export class QuoteHandler extends APIGLambdaHandler<
   }
 
   protected responseBodySchema(): Joi.ObjectSchema | null {
-    return null;
+    return HardQuoteResponseDataJoi;
+  }
+}
+
+export function getCosignerData(request: HardQuoteRequest, quote: QuoteResponse): CosignerData {
+  const decayStartTime = getDecayStartTime(request.tokenInChainId);
+  // default to open order with the original prices
+  let filler = ethers.constants.AddressZero;
+  let inputAmount = BigNumber.from(0);
+  const outputAmounts = request.order.info.baseOutputs.map(() => BigNumber.from(0));
+
+  // if the quote is better, then increase amounts by the difference
+  if (request.type === TradeType.EXACT_INPUT) {
```

Comment:
What's the expected behavior if less outputs are required for EXACT_INPUT but more input is required? Apply the output adjustment but not the input? Wonder if we should explicitly throw / error on this case if its undesired behavior

Reply:
Hmm interesting, I think the api contract should disallow changing the EXACT_X side in any case yea